stone/ir/data_types.py

```diff
@@ -1933,6 +1933,31 @@ def unwrap_aliases(data_type):
         data_type = data_type.data_type
     return data_type, unwrapped_alias
 
+def resolve_aliases(alias):
+    """
+    Resolve all chained / nested Aliases - this will recursively point
+    nested aliases to their resolved data type (first non-alias in the chain).
+
+    Note: This differs from unwrap_alias which simply identifies/returns
+    the resolved data type.
+
+    Args:
+        alias (Alias): The target Alias to resolve.
+    Return:
+        Alias: The resolved parent object.
+    """
+    def _resolve_data_type_chain(data_type):
+        if not is_alias(data_type):
+            return data_type
+
+        resolved = _resolve_data_type_chain(data_type.data_type)
+        data_type.data_type = resolved
+
+        return resolved
+
+    alias.data_type = _resolve_data_type_chain(alias.data_type)
+
+    return alias
```

Comment:
I'm thinking if we just return whatever alias passed in, it's not something quite useful. Should we simply return the resolved data type (as stated in the comment on line 1947)?. If we change it this way, we could avoid the internal function `_resolve_data_type_chain` to make the code cleaner.

Reply:
Okay I'll just revert to my original implementation.